fedavg.py

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+import tensorflow as tf
+from tensorflow_federated.python.learning import federated_averaging
+import os
+import asyncio
+from fastapi import FastAPI, WebSocket, File, UploadFile
+from fastapi.responses import HTMLResponse
+import websockets
+import copy
+
+app = FastAPI()
+
+datasets = []
+num_clients = 1
+model = None
+clients = [0] #this would be a list of all client ids
+
+@app.get('/')
+async def root:
+    return 'root'
+
+@app.get("/{model}/")
+async def update_model(model: tf.keras.Model):
+    return model.load_weights('consolidated.h5')
+
+@app.post("/weights")
+async def load_weights(image: UploadFile = File(...)):
+    model.load_weights(image.filename)
+    datasets.append(copy.deepcopy(model))
+    load_data()
+    return {'consolidated weights': model}
+
+async def load_data():
```

Comment:
this function is calling `load_weights`, while `load_weights` calls this function. this would not work. 

i think you just want to call `fed_avg` in `load_weights` for now.